starlite/app.py

```diff
@@ -603,8 +603,6 @@ def _create_asgi_handler(self) -> "ASGIApp":
             asgi_handler = TrustedHostMiddleware(app=asgi_handler, allowed_hosts=self.allowed_hosts)  # type: ignore
         if self.cors_config:
             asgi_handler = CORSMiddleware(app=asgi_handler, **self.cors_config.dict())  # type: ignore
-        if self.csrf_config:
-            asgi_handler = CSRFMiddleware(app=asgi_handler, config=self.csrf_config)
```

Comment:
I had to move CSRFMiddleware application to the route handler level (` _build_route_middleware_stack` function) because at this point of execution route is not resolved yet.